### PR TITLE
Main navigation upgrade to Ant Design

### PR DESCRIPTION
**Pull request by joshsadam**
## Description of changes

Screen Shot 2020-11-26 at 1 20 15 PM

Update main navigation to use Ant Design.  In the process I added the ability to customize any Ant Design properties found in the [custom variables less file](https://github.com/ant-design/ant-design/blob/master/components/style/themes/default.less).  To do this just take the less variable e.g. `@primary-color` remove the `@` and prepend `styles.ant.` to it --> `styles.ant.primary-color` and add it either to `/etc/irida/irida.conf` file or to `src/main/resources/configuration.properties`.  Light or dark themes can also be set by setting `styles.theme=` to either `light` or `dark`.

E.g. Adding:
```
styles.ant.primary-color=#eb2f96
styles.ant.info-color=#1890ff
styles.ant.link-color=#1890ff
styles.ant.font-size-base=14px
styles.ant.border-radius-base=10px
```
Will result in IRIDA looking like:
Screen Shot 2020-11-26 at 1 10 50 PM

## Related issue
N/A

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [x] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
* [x] Tests added (or description of how to test) for any new features.
* [x] User documentation updated for UI or technical changes.

### CODE REVIEW

File: src/main/webapp/styles.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+};
+const iridaConfig = "/etc/irida/irida.conf";
+const propertiesConfig = "../resources/ca/corefacility/bioinformatics/irida/config/styles.properties";
+
+function formatAntStyles() {
+  const colourProperties = {};
+  const re = /styles.ant.([\w+-]*)/;
+
+  try {
+    if (fs.existsSync(propertiesConfig)) {
```
Comment (deepsidhu85) [inline src/main/webapp/styles.js:19]: The user can set the properties in irida.conf or styles.properties correct? Do we want to allow the user to mix and put some styles in both files? I think this `if` and the `if` blocl below should be changed to an if/else block

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
Just a few changes/comments

File: src/main/webapp/resources/js/components/Header/MainNavigation.jsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+              <a
+                href="http://www.irida.ca"
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                {i18n("generic.irida.website")}
+              </a>
+            </Menu.Item>
+            <Menu.Divider />
+            <Menu.Item disabled>{i18n("irida.version")}</Menu.Item>
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/components/Header/MainNavigation.jsx:131]: This is the way it is in the current one, I need to keep it there.

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/security/LoginSuccessHandler.java [hunk 3]
```diff
@@ -53,6 +67,10 @@ public void onAuthenticationSuccess(HttpServletRequest httpServletRequest, HttpS
 		// Add the user into the session
 		session.setAttribute("user", user);
 
+		// Add the site them
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/security/LoginSuccessHandler.java:70]: Fixed in 25f217a1a3e37209c3e9801072b6e89209b1abb8

File: src/main/webapp/resources/js/components/Header/MainNavigation.jsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+          />
+        </a>
+        <Menu mode="horizontal" theme={theme}>
+          <Menu.SubMenu key="projects" title={i18n("nav.main.project")}>
+            <Menu.Item key="project:list">
+              <a href={setBaseUrl("/projects")}>
+                {i18n("nav.main.project-list")}
+              </a>
+            </Menu.Item>
+            {isAdmin && (
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/components/Header/MainNavigation.jsx:36]: It is basically a shortcut operator in this case for a ternary.  There are some cases where this is bad, but it works here.

[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/components/Header/MainNavigation.jsx:36]: in other words it is equibalent to `isAdmin ? <Compenonet .../> : null}`

File: src/main/webapp/resources/js/components/Header/MainNavigation.jsx [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+          </Menu.SubMenu>
+          {isAdmin && (
+            <Menu.Item key="admin:link">
+              <a className="t-admin-panel-btn" href={setBaseUrl("/admin")}>
+                {i18n("MainNavigation.admin").toUpperCase()}
+              </a>
+            </Menu.Item>
+          )}
+          <Menu.SubMenu title={<IconQuestionCircle />}>
+            <Menu.ItemGroup title="Documentation">
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/components/Header/MainNavigation.jsx:103]: Updated in ffccf964c6a7a2c0bd00749a09fb478d9abebc85

File: src/main/webapp/styles.js (hunk 1 above)
Comment (joshsadam) [reply src/main/webapp/styles.js:19]: Updated in 264b0e5cc9ce579cf439f25630edddc6cf4595f5

File: src/main/webapp/resources/js/components/Header/MainNavigation.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/Header/MainNavigation.jsx:131]: I will give you a pass on this one :)

File: src/main/webapp/resources/js/components/Header/MainNavigation.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/Header/MainNavigation.jsx:36]: Awesome thanks

File: src/main/webapp/resources/js/components/Header/MainNavigation.jsx [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+              </a>
+            </Menu.Item>
+          </Menu.SubMenu>
+        </Menu>
+      </Col>
+      <Col
+        md={14}
+        sm={24}
+        style={{
+          display: "flex",
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/Header/MainNavigation.jsx:76]: As discussed we will think on this for a later PR if need be

**Review by deepsidhu85 — APPROVED** [review]

**Pull request merged**